gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingExtractor.java

```diff
@@ -203,15 +210,17 @@ public S getSchema() throws IOException {
    * Return the next record when available. Will never time out since this is a streaming source.
    */
   @Override
-  public RecordEnvelope<D> readRecordImpl(RecordEnvelope<D> reuse) throws DataRecordException, IOException {
+  public RecordEnvelope<D> readRecordImpl(RecordEnvelope<D> reuse)
+      throws DataRecordException, IOException {
     if (!_isStarted.get()) {
       throw new IOException("Streaming extractor has not been started.");
     }
     while ((_records == null) || (!_records.hasNext())) {
       synchronized (_consumer) {
-        if (_close.get())
+        if (_close.get()) {
           throw new ClosedChannelException();
-        _records = _consumer.poll(100).iterator();
+        }
+        _records = _consumer.poll(10000).iterator();
```

Comment:
This should be configurable. Maybe use a key similar to the one in AbstractBaseKafkaConsumerClient. CONFIG_KAFKA_FETCH_TIMEOUT_VALUE.